xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +80,43 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights and updates MSet.
+     *
+     *  Dereferencing the Iterator should return a double.
+     *
+     *  Iterator should dereference weights in the original order of documents.
+     *  Weights are assigned as per the original order of documents.
+     *
+     *  @param begin	Begin iterator.
+     *  @param end	End iterator.
+     *
+     *  @exception Xapian::InvalidArgument is thrown if the total number of
+     *		    elements in the input doesn't match the total number of
+     *		    documents in MSet.
+     */
+    template <typename Iterator>
+    void replace_weights(Iterator first, Iterator last)
+    {
+	if (last - first != size()) {
+	    throw Xapian::InvalidArgumentError("Number of weights assigned"
+					       " doesn't match the number of "
+					       "items.");
```

Comment:
I'd suggest consistently breaking the string after a space, as that's how we usually do it (currently you have one split before then one after).

Also, we don't usually include a full-stop on an exception message.  It's not entirely consistent currently, but a quick grep suggests it's about 94% without.